openhcl/virt_mshv_vtl/src/processor/mshv/x64.rs

```diff
@@ -1905,6 +1900,65 @@ impl<T> hv1_hypercall::SetVpRegisters for UhHypercallHandler<'_, '_, T, Hypervis
     }
 }
 
+impl<T> hv1_hypercall::ModifyVtlProtectionMask
+    for UhHypercallHandler<'_, '_, T, HypervisorBackedX86>
+{
+    fn modify_vtl_protection_mask(
+        &mut self,
+        partition_id: u64,
+        _map_flags: HvMapGpaFlags,
+        target_vtl: Option<Vtl>,
+        gpa_pages: &[u64],
+    ) -> hvdef::HvRepResult {
+        if partition_id != hvdef::HV_PARTITION_ID_SELF {
+            return Err((HvError::AccessDenied, 0));
+        }
+
+        let target_vtl = self
+            .target_vtl_no_higher(target_vtl.unwrap_or(self.intercepted_vtl.into()))
+            .map_err(|e| (e, 0))?;
+        if target_vtl == GuestVtl::Vtl0 {
+            return Err((HvError::InvalidParameter, 0));
+        }
+
+        // A VTL cannot change its own VTL permissions until it has enabled VTL protection and
+        // configured default permissions. Higher VTLs are not under this restriction (as they may
+        // need to apply default permissions before VTL protection is enabled).
+        if target_vtl == self.intercepted_vtl {
```

Comment:
(Not introduced by your change) Is there ever a case where this is false? I can only think of target_vtl==1, intercepted_vtl==0, but it seems that that's disallowed by the `target_vtl_no_higher` check above. target=0, intercepted=1 is disallowed by the `if target_vtl == GuestVtl::Vtl0` check, correct?

Reply:
Yeah that's true. In which case the only benefit to keeping it here is if we introduce higher vtls, or to make sure that anyone looking at this code understands the requirements.

Reply:
Having said that, I think I'm in favor of keeping it as-is: the code explains what's going on, and if we took out the `if` there, we'd definitely want a big comment explaining it.